lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,261 @@
+  state: present
+'''
+
+RETURN = '''
+'''
+
+import os
+
+try:
+    from OpenSSL import crypto
+except ImportError:
+    pyopenssl_found = False
+else:
+    pyopenssl_found = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.crypto import OpenSSLModule, OpenSSLModuleError, load_certificate, load_privatekey
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils._text import to_native
+
+
+class PkcsError(OpenSSLModuleError):
+    pass
+
+
+class Pkcs(OpenSSLModule):
+
+    def __init__(self, module):
+        super(Pkcs, self).__init__(module)
+        self.action = module.params['action']
+        self.iter_size = module.params['iter_size']
+        self.maciter_size = module.params['maciter_size']
+        self.privatekey_path = module.params['privatekey_path']
+        self.privatekey_passphrase = module.params['privatekey_passphrase']
+        self.cert_path = module.params['cert_path']
+        self.ca_certificates = module.params['ca_certificates']
+        self.friendly_name = module.params['friendly_name']
+        self.passphrase = module.params['passphrase']
+
+    def export(self):
+        ''' Generate pkcs#12 file archive. '''
+
+        if os.path.exists(self.path) and not self.force:
+            self.changed = False
+            return
+
+        self.pkcs12 = crypto.PKCS12()
+
+        if self.ca_certificates:
+            ca_certs = tuple([load_certificate(ca_cert) for ca_cert in self.ca_certificates])
+            self.pkcs12.set_ca_certificates(ca_certs)
+
+        if self.cert_path:
+            self.pkcs12.set_certificate(load_certificate(self.cert_path))
+
+        if self.friendly_name:
+            self.pkcs12.set_friendlyname(self.friendly_name)
+
+        if self.privatekey_path:
+            self.pkcs12.set_privatekey(load_privatekey(
+                                        self.privatekey_path,
+                                        self.privatekey_passphrase)
+                                        )
+
+        try:
+            with open(self.path, 'w') as archive:
+                archive.write(
+                    self.pkcs12.export(
+                        self.passphrase,
+                        self.iter_size,
+                        self.maciter_size
+                    )
+                )
+        except (IOError, OSError) as exc:
+            raise PkcsError(exc)
+
+    def check(self):
+        return True
+
+    def dump(self):
+        result = {
+            'changed': self.changed,
+            'path': self.path,
+        }
+
+        return result
+
+
+def main():
+    argument_spec = dict(
+        action=dict(default='export',
+                    choices=['parse', 'export'],
+                    type='str'),
+        ca_certificates=dict(type='list'),
+        cert_path=dict(type='path'),
+        force=dict(default=False, type='bool'),
+        friendly_name=dict(required=True, type='str'),
```

Comment:
According to the documentation `aliases` keyword is missing for `friendly_name`.